### PR TITLE
can parse fixture ACL in ttl and hasUser function works 

### PR DESCRIPTION
**Pull request by ndushay**
this also removes some placeholder code that was originally needed to get the CI build working.

connects to #3

### CODE REVIEW

**Comment by ndushay** [comment]
Note codeclimate threshold chosen was arbitrary;  I couldn't even see the coverage in codeclimate until after this PR.

File: __tests__/webAccessControl.test.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { WebAccessControl } from '../src/webAccessControl'
+const fs = require('fs')
```
Comment (mjgiarlo) [inline __tests__/webAccessControl.test.js]: Would you consider `import fs from 'fs'` here? That way this test would use a consistent method for loading code.

File: __tests__/webAccessControl.test.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+describe('WebAccessControl', () => {
+  const myWac = new WebAccessControl()
+  const fixture_dir = '__tests__/__fixtures__'
+
+  describe('parseWac', () => {
+    it('errors when wac is jsonld', () => {
+      expect(() => {
+        myWac.parseWac(fs.readFileSync(`${fixture_dir}/defaultBaseAcl.jsonld`).toString())
```
Comment (mjgiarlo) [inline __tests__/webAccessControl.test.js]: No problem with this, but here's an alternative I just learned about. If you pass the second arg to `fs.readFileSync`, for an encoding, it automatically returns a string. If you omit it, you get a buffer. I'm not sure it matters here, though.

File: __tests__/webAccessControl.test.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      }).toThrow('Expected entity but got literal')
+    })
+    it('no error when the arg is an empty string', () => {
+      myWac.parseWac('')
```
Comment (mjgiarlo) [inline __tests__/webAccessControl.test.js]: If this did throw, would the test pass? I guess it'd error out, and that's as good as a fail? (Wondering whether we need an explicit expectation for these two tests or whether this is good for now. Prolly the latter.)

File: src/webAccessControl.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
+// Simplified model of WebAccessControl (https://www.w3.org/wiki/WebAccessControl)
+// At this time:
+// - everything in Sinopia Server (Trellis LDP server) is world readable
+// - We are only concerned with permissions for "group containers", which are ldp:BasicContaier
+// - We expect each group container to have explicit permissions, rather than relying on default permissions of ldp:BaseContainer
+// - If a user is given write access for a group, then they get all of acl:Read, acl:Write, and acl:Control
```
Comment (mjgiarlo) [inline src/webAccessControl.js]: Makes sense to me. Thanks for adding the docs!

File: src/webAccessControl.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+const N3 = require('n3')
+const { DataFactory } = N3
+// const { namedNode, literal, defaultGraph, quad } = DataFactory
+const { namedNode } = DataFactory
+
+export class WebAccessControl {
+
+  constructor(wacData=null) {
+    this.n3store = N3.Store()
+    if (wacData != null && wacData.length > 5)
```
Comment (mjgiarlo) [inline src/webAccessControl.js]: What's the significance of the `5` here?

File: src/webAccessControl.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  // }
+
+  // expect userid to be a string
+  hasUser(userid) {
+    // TODO: memoize userNodeArray for instance?
+    const userNodeArray = this.n3store.getObjects(null, namedNode('http://www.w3.org/ns/auth/acl#agent'), null)
+    if (userNodeArray == undefined || userNodeArray.size == 0)
+      return false
+
+    var matchingNode = userNodeArray.find((element) => {
```
Comment (mjgiarlo) [inline src/webAccessControl.js]: Minor preference to use `let` over `var`, but for all I know we could have `var` all over the place, so maybe no bigs.

Comment (mjgiarlo) [reply src/webAccessControl.js]: I'm on the fence about this, but how would you feel about using something like:


```javascript
return userNodeArray.map(element => element.value).includes(userid)
```

I'm not sure which is more readable. What do you think?

(With this approach you can also remove the `|| userNodeArray.size == 0` check above.)

File: src/webAccessControl.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  }
+
+  // userCanWrite(userid) {
+  //
+  // }
+
+  // expect wacData to be a string
+  parseWac(wacData) {
+    const parser = N3.Parser()
+    this.n3store = N3.Store()
```
Comment (mjgiarlo) [inline src/webAccessControl.js]: Does this mean every time we call `parseWac`, we expect `this.n3store` to be reinitialized?

I'm wondering about this class having multiple entrypoints for initiating a parse, one in the constructor and one being this method. Could we simplify by reducing down to one parsing entrypoint or am I blanking on a use case for multiple?

File: src/webAccessControl.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    if (wacData)
+      this.n3store.addQuads(parser.parse(wacData))
```
Comment (mjgiarlo) [inline src/webAccessControl.js]: Is there an expectation of a certain sort of return value from this function? or do we use it solely for the effect of modifying `this.n3store`?

File: __tests__/webAccessControl.test.js (hunk 2 above)
Comment (ndushay) [reply __tests__/webAccessControl.test.js]: hrm.  tried it, couldn't get it working with args:

`utf8`
`utf-8`
`{encoding: "utf8"}`
`{encoding: "utf-8"}`

File: __tests__/webAccessControl.test.js (hunk 3 above)
Comment (ndushay) [reply __tests__/webAccessControl.test.js]: yes, if it throws an error that isn't trapped, the test fails.  This is ok for now.

File: src/webAccessControl.js (hunk 5 above)
Comment (ndushay) [reply src/webAccessControl.js]: I was trying to think of the shortest triple i could imagine in a ttl doc.  `a b c`.   but it is arbitrary.

File: src/webAccessControl.js (hunk 6 above)
Comment (ndushay) [reply src/webAccessControl.js]: sure!  I made this change - thanks for the suggestion.   I just got sick of farting around and wanted to get code that did what I knew it should with tests that pass.  This is more elegant and more rubyish and I like it fine.

File: src/webAccessControl.js (hunk 7 above)
Comment (ndushay) [reply src/webAccessControl.js]: my intent was to make it possible to:

1. pass in wacData to the constructor, thereby avoiding making an explicit call to parseWac
2. not pass in wacData to the constructor and allow parseWac to be called later

note that this way also affords:
3. re-use the instance if desired, by explicitly calling parseWac

I'm not tied to my approach - again, it was a way to just get SOMETHING working.  How important is this to you?

File: src/webAccessControl.js (hunk 8 above)
Comment (ndushay) [reply src/webAccessControl.js]: I was thinking we didn't care about a return value - the important thing was to get the data into the instance so we could work with it.

**Comment by ndushay** [comment]
@mjgiarlo I have tweaked - let me know if you are still uncomfy with anything.

File: __tests__/webAccessControl.test.js (hunk 2 above)
Comment (mjgiarlo) [reply __tests__/webAccessControl.test.js]: @ndushay I believe `'utf8'` is what I saw work as second arg but it's no biggie in this context. You've got me!

File: src/webAccessControl.js (hunk 5 above)
Comment (mjgiarlo) [reply src/webAccessControl.js]: @ndushay ok. so it's a sort of validation?

File: src/webAccessControl.js (hunk 7 above)
Comment (mjgiarlo) [reply src/webAccessControl.js]: @ndushay OK, I getcha. Mostly I was thinking about reducing code complexity (by removing these lines from the constructor and forcing one usage mode on this class), along YAGNI lines. I don't feel strongly about it! We should get this merged and moving.

File: src/webAccessControl.js (hunk 8 above)
Comment (mjgiarlo) [reply src/webAccessControl.js]: okie dokie.

File: src/webAccessControl.js (hunk 5 above)
Comment (ndushay) [reply src/webAccessControl.js]: yeah, i guess.  like, if it's empty, there's no point in parsing it.

**Pull request merged**